codegen/gql_code_builder/lib/src/inline_fragment_classes.dart

```diff
@@ -94,28 +97,35 @@ List<Spec> buildInlineFragmentClasses({
         // print("alias $typeName => ${dataClassAliasMap[typeName]!.symbol}");
         return false;
       }
+      // is it okay to inlcude interfaces?
       return true;
     }).expand(
       (inlineFragment) => buildSelectionSetDataClasses(
-          name: "${name}__as${inlineFragment.typeCondition!.on.name.value}",
-          selections: mergeSelections(
-            [
-              ...selections.whereType<FieldNode>(),
-              ...selections.whereType<FragmentSpreadNode>(),
-              ...inlineFragment.selectionSet.selections,
-            ],
-            fragmentMap,
-          ),
-          fragmentMap: fragmentMap,
-          dataClassAliasMap: dataClassAliasMap,
-          schemaSource: schemaSource,
-          type: inlineFragment.typeCondition!.on.name.value,
-          typeOverrides: typeOverrides,
-          superclassSelections: {
-            name: SourceSelections(url: null, selections: selections)
-          },
-          built: built,
-          whenExtensionConfig: whenExtensionConfig),
+        name: "${name}__as${inlineFragment.typeCondition!.on.name.value}",
+        selections: mergeSelections(
+          [
+            ...selections.whereType<FieldNode>(),
+            ...selections.whereType<FragmentSpreadNode>(),
+            ...inlineFragment.selectionSet.selections,
+          ],
+          fragmentMap,
+        ),
+        fragmentMap: fragmentMap,
+        possibleTypesMap: possibleTypesMap,
+        dataClassAliasMap: dataClassAliasMap,
+        schemaSource: schemaSource,
+        type: inlineFragment.typeCondition!.on.name.value,
+        typeOverrides: typeOverrides,
+        superclassSelections: {
+          name: SourceSelections(url: null, selections: selections),
+          if (dataClassAliasMap.isNotEmpty)
```

Comment:
I think this condition is not correct.
Inline Fragments can add new fields in selections which can cause conflicts in extending the fragments. 

It also causes the build error in the end to end test package.

When Removing this, it works again